scripts/TestGen.hx

```diff
@@ -0,0 +1,380 @@
+
+		var renderedMethods = new Array<String>();
+		for (key in methods.keys())
+		{
+			var method = methods.get(key);
+			renderedMethods.insert(method.order, method.method);
+		}
+
+		for (i in 0...stories.length)
+		{
+			stories[i] = storyTmpl.execute(
+				{
+					story: stories[i],
+					testCases: testCases.get(stories[i]).join("")
+				});
+		}
+
+		// create String array for README; add exercise as title
+		var readme = new Array<String>();
+		readme.push('# ${toUpperCamel(exercise, " ")}');
+		// Copy descriptionMd to README
+		readme.push(descriptionMd);
+		// extremely naive metadata yaml parsing
+		var splitYml = metaToml.split("\n");
+		if (splitYml.length > 3)
+		{
+			readme.push('## Source');
+			readme.push(splitYml[2].split("source =")[1].replace("\"", "").trim());
+			readme.push(splitYml[3].split("source_url =")[1].replace("\"", "").trim());
+		}
+		// Add incomplete blurb
+		readme.push("\n## Submitting Incomplete Solutions\nIt's possible to submit an incomplete solution so you can see how others have completed the exercise.");
+
+		// Create README file
+		Sys.println("Creating README");
+		File.saveContent('${dir}/README.md', readme.join("\n"));
+
+		// create .meta/tests.toml
+		Sys.println("Creating .meta/tests.toml");
+		File.saveContent('${dir}.meta/tests.toml', meta.join(""));
+
+		// create Exercise.hx stub file
+		var exercisePath = 'src/${toUpperCamel(exercise)}.hx';
+		Sys.println("Creating " + exercisePath);
+		File.saveContent('${dir}${exercisePath}', exerciseStubTmpl.execute(
+			{
+				exercise: toUpperCamel(exercise),
+				methods: renderedMethods.join("\n\n\t")
+			}));
+
+		// create empty Example.hx
+		Sys.println('Creating .meta/Example.hx');
+		File.saveContent('${dir}.meta/Example.hx', exerciseStubTmpl.execute(
+			{
+				exercise: toUpperCamel(exercise),
+				methods: renderedMethods.join("\n\n\t")
+			}));
+
+		// create Test.hx file
+		Sys.println('Creating test/Test.hx');
+		File.saveContent('${dir}test/Test.hx', testTmpl.execute(
+			{
+				stories: stories.join("\n\t\t")
+			}));
+
+		// copy test.hxml from hello-world
+		Sys.println('Copying test.hxml');
+		File.copy('${exercisesDir}hello-world/test.hxml', '${dir}test.hxml');
+
+		Sys.println("Generation Successful");
+		Sys.println("Don't forget to update config.json!");
+	}
+
+	private static function generateTests(cases:Array<Dynamic>, methods:Map<String, {order:Int, method:String}>, stories:Array<String>,
+			testCases:Map<String, Array<String>>, meta:Array<String>, ?story:String)
+	{
+		for (testCase in cases)
+		{
+			if (testCase.cases == null)
+			{
+				// if story does not exist make a new one
+				if (story == null)
+				{
+					story = toUpperCamel(exercise, " ");
+				}
+				if (story != null && !testCases.exists(story))
+				{
+					stories.push(story);
+					testCases.set(story, new Array<String>());
+				}
+
+				// add description to .meta/test.toml
```

Comment:
There's no need to edit tests.toml. It's an auto-generated file so any manual edits will be lost anyways.